src/nvim/lua/vim.lua

```diff
@@ -1,2 +1,62 @@
 -- TODO(ZyX-I): Create compatibility layer.
-return {}
+--{{{1 package.path updater function
+-- Last inserted paths. Used to clear out items from package.[c]path when they
+-- are no longer in &runtimepath.
+local last_nvim_paths = {}
+local function _update_package_paths()
+  local new_last_nvim_paths = {}
+  local rtps = vim.api.nvim_list_runtime_paths()
+  for _, key in ipairs({'path', 'cpath'}) do
+    local orig_str = package[key] .. ';'
+    local pathtrails = {}
+    local pathtrails_ordered = {}
+    local orig = {}
+    local orig_paths = {}
+    -- Note: ignores trailing item without trailing `;`. Not using something
+    -- simpler in order to preserve empty items (stand for default path).
+    for s in orig_str:gmatch('[^;]*;') do
+      s = s:sub(1, -2)  -- Strip trailing semicolon
+      orig[#orig + 1] = s
+      orig_paths[s] = true
+      -- Find out path patterns. pathtrail should contain something like
+      -- /?.so, /?/init.lua, /?.lua. This allows not to bother determining what
+      -- correct suffixes are.
+      local pathtrail = s:match('[/\\][^/\\]*%?.*$')
+      if pathtrail and not pathtrails[pathtrail] then
+        pathtrails[pathtrail] = pathtrail:sub(1, 1)
```

Comment:
E.g.: pathtrails will only contain the (first) path separator of pathtrail (example: `/?.lua`), which may only be slash or backslash?

I'm curious why the separator is determined for every path. Should they not be the same over the entire system? Which systems have this mixed mode?

Relatedly, why cache this in pathtrails? The name is not indicative of the fact that it's a map from pathtrail to its own first separator. Calculating it on the fly (with a small) comment down in the `rtp:match(';')` block would do just as well, I think.